synfig-studio/src/gui/mainwindow.cpp

```diff
@@ -194,6 +194,17 @@ MainWindow::init_menus()
 		sigc::ptr_fun(App::edit_custom_workspace_list)
 	);
 
+	//animation tabs
+	for (int i = 1; i <= 8; ++i) {
+		const std::string tab = std::to_string(i);
+		action_group->add(Gtk::Action::create("switch-to-tab-" + tab, _("Switch to Tab ") + tab),
+			[this, i]() { main_dock_book().set_current_page(i-1); }
```

Comment:
Remember that issue with signal slots that are lambda functions?
Protect them  (`sigc::track_obj`) or change to a class member function (`sigc::mem_fun()`)

Reply:
oh yes of course, done ^_^.